### PR TITLE
test: add unity test framework

### PR DESCRIPTION
**Pull request by jsarzy**

### CODE REVIEW

File: test/bsearch/main.c [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+{
+}
+
+TEST_TEAR_DOWN(test_bsearch)
+{
+}
+
+TEST(test_bsearch, basic)
+{
+	int arr_inc[] = {1, 2, 3, 4, 5, 6, 7, 8};
```
Comment (nalajcie) [inline test/bsearch/main.c:38]: `const int` (the same apples everywhere else)

File: test/bsearch/main.c [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	int arr_dec[] = {7, 6, 5, 4, 3, 2, 1};
+	int arr_single[] = {1};
+
+	for (int key = 1; key < 8; key++) {
```
Comment (nalajcie) [inline test/bsearch/main.c:42]: `<=`, You're never searching for `8`

File: test/bsearch/main.c [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+		void *res = lib_bsearch(&key, arr_inc, 8, sizeof(int), cmp_gt);
```
Comment (nalajcie) [inline test/bsearch/main.c:43]: `sizeof(arr_inc) / sizeof(arr_inc[0])`, better than magic number

File: test/bsearch/main.c [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+		TEST_ASSERT_EQUAL_PTR(&arr_inc[key-1], res);
+	}
+
+	for (int key = 7; key > 1; key--) {
```
Comment (nalajcie) [inline test/bsearch/main.c:47]: never searching for `1`

File: test/sched/main.c [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+	actual_prio = -1;
+
+	for (unsigned int p = 0; p < THREADS_MAX_PRIO; p++) {
+		int res = proc_threadCreate(NULL, thread_prio, NULL, p, stacksz, NULL, 0, (void *)p);
+		TEST_ASSERT_EQUAL_INT(EOK, res);
+	}
+
+	for (unsigned int p = 0; p < THREADS_MAX_PRIO; p++) {
+		proc_reap();
+	}
```
Comment (nalajcie) [inline test/sched/main.c:42]: maybe we could check `actual_prio` value at the test end also?

File: test/test_main.c [hunk 6]
```diff
@@ -0,0 +1,24 @@
+#include "test.h"
+#include "proc/threads.h"
+
+DECLARE_TEST_GROUP(test_list);
```
Comment (nalajcie) [inline test/test_main.c:4]: we should probably comment that it's necessary to explicitly declare new test groups here

File: test/Makefile [hunk 7]
```diff
@@ -5,5 +5,21 @@
...
+KERNEL_TEST_GROUPS := $(subst $(COMMA),$(SPACE),$(KERNEL_TEST_GROUPS))
+
+CFLAGS += -DKERNEL_TEST
+CFLAGS += -I$(PREFIX_H)unity
+
+UNITYLIB := $(PREFIX_A)unity.a
+
+OBJS += $(PREFIX_O)test/test_main.o
+
+include $(addprefix test/, $(addsuffix /Makefile, $(KERNEL_TEST_GROUPS)))
```
Comment (nalajcie) [inline test/Makefile:21]: I see that using `,` as a separator is quite confusing in `Makeflie`. We can leave it as is, but I would welcome some way to compile and run all tests (would `KERNEL_TEST_GROUPS=*` work?)

**Review by nalajcie — CHANGES_REQUESTED** [review]

File: test/bsearch/main.c (hunk 1 above)
Comment (kemonats) [reply test/bsearch/main.c:38]: better `static const int`

**Pull request closed without merging**